File: Java_ChildrenCompetition/src/main/java/service/IParticipantService.java
package service;

import model.Participant;

import java.util.List;

//public interface IParticipantService  <ID, T extends HasId<ID>>{
//    void save(String username, String name, int age, int testId);
//    T findByUsername(String username);
//    List<T> findAllParticipantsForTest(Integer id);
//}

public interface IParticipantService {
    void save(String username, String name, int age, int testId);
    Participant findByUsername(String username);
    List<Participant> findAllParticipantsForTest(Integer id);
}

File: CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/service/IParticipantService.cs
using System.Collections.Generic;
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.service
{
    public interface IParticipantService<ID, T> where T : HasId<ID>
    {
        void save(string username, string name, int age, int testId);
        T findByUsername(string username);
        List<T> findAllParticipantsForTest(int id);
    }
}